src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -1512,12 +1499,12 @@ namespace
         //                   params[P_0, P_1, .. P_(A-1), indices[I_0, .., I_(M-1)],
         //                          P_(A+1), ... P_(N-1)];
 
-        AffineLoopNestBuilder(indicesIVPtrs, indicesLbs, indicesUbs, indicesSteps)([&] {
+        AffineLoopNestBuilder(indicesIVs, indicesLbs, indicesUbs, indicesSteps)([&] {
             // Load axis value from indices array and cast it to Index Type
-            ValueHandle axisIdx = ValueHandle::create<IndexCastOp>(
-                (ValueHandle)iIndices(indicesIVs), rewriter.getIndexType());
+            Value axisIdx =
+                ValueBuilder<IndexCastOp>((Value)iIndices(indicesIVs), rewriter.getIndexType());
```

Comment:
is the Value casting still necessary? There are plenty of them in the code. Maybe they can be removed.

Reply:
Looks like it can be removed.  Good catch.